### ISSUE TITLE
MenuFlyout causes application crash

### ISSUE THREAD

**Issue by JozefGula** [microsoft/microsoft-ui-xaml#8954]
### Describe the bug

Hello, if I add this line to the project, opening any MenuFlyout causes a crash.

```
<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>
```

This line is necessary to publish a functional application, but no FlayoutMenu works with it.

The dump file contains:

```
ERROR_CODE: (NTSTATUS) 0xc000027b - An application-internal exception has occurred.

EXCEPTION_CODE_STR:  80040154

EXCEPTION_PARAMETER1:  0000021a668f0850

EXCEPTION_PARAMETER2:  0000000000000001

FAULTING_THREAD:  ffffffff

STACK_TEXT:  
0000021a`66969010 00007ff8`e8d4ec1e Microsoft_ui_xaml!CPopup::EnsureContentExternalBackdropLink+0xa2
0000021a`66969018 00007ff8`e8c6beb2 Microsoft_ui_xaml!CPopup::SetValue+0x262
0000021a`66969020 00007ff8`e8b234ec Microsoft_ui_xaml!DirectUI::DependencyObject::SetValueCore+0x2fc
0000021a`66969028 00007ff8`e8c7215e Microsoft_ui_xaml!DirectUI::DependencyObject::SetValueByKnownIndex+0x8e
0000021a`66969030 00007ff8`e8d5a33a Microsoft_ui_xaml!DirectUI::FlyoutBase::ForwardSystemBackdropToPopup+0x92
0000021a`66969038 00007ff8`e930fb75 Microsoft_ui_xaml!DirectUI::FlyoutBase::Open+0x411
0000021a`66969040 00007ff8`e9311c3e Microsoft_ui_xaml!DirectUI::FlyoutBase::ShowAtCore+0x15e
0000021a`66969048 00007ff8`e931c55c Microsoft_ui_xaml!DirectUI::MenuFlyout::ShowAtCore+0xec
0000021a`66969050 00007ff8`e931c43d Microsoft_ui_xaml!DirectUI::MenuFlyout::ShowAt+0x3d


SYMBOL_NAME:  Microsoft_ui_xaml!CPopup::EnsureContentExternalBackdropLink+a2

MODULE_NAME: Microsoft_ui_xaml

IMAGE_NAME:  Microsoft.ui.xaml.dll

STACK_COMMAND:  *** Stowed Exception v2 ***; .exr -1; dpp 0x21a668f0850 L0x1; dt 0x21a6696a028 combase!STOWED_EXCEPTION_INFORMATION_V2 -r; dps 0x21a66969010 L0xa ; ** Pseudo Context ** StowedPseudo ** Value: 3 ** ; kb

FAILURE_BUCKET_ID:  STOWED_EXCEPTION_80040154_Microsoft.ui.xaml.dll!CPopup::EnsureContentExternalBackdropLink

OS_VERSION:  10.0.19041.1

BUILDLAB_STR:  vb_release

OSPLATFORM_TYPE:  x64

OSNAME:  Windows 10

IMAGE_VERSION:  3.1.4.0

FAILURE_ID_HASH:  {d2aaf439-c56d-2a02-3fb2-e2b0861c3d2d}

Followup:     MachineOwner
```

I found an old issue, but it refers to a fix in a version that is older than the one I use: https://github.com/microsoft/microsoft-ui-xaml/issues/8744

My version of "Microsoft.WindowsAppSDK" is "1.4.230913002.

### Steps to reproduce the bug

1. Add WindowsAppSDKSelfContained to project
2. Trying to open any FlayoutMenu causes a crash

it is possible that this problem is directly related to some packages in my application. If necessary, I can provide my entire project, but not for public sharing. If necessary, I will try to reproduce this problem on a simpler application.

### Expected behavior

1. MenuFlyout works without this line in the project, but it does not work with this line in my project
2. My app does not work after publishing without this line in the project, but it works with this line in the project

These two things are mutually exclusive and I need both the application and the MenuFlyout to work. 

### Screenshots

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/8478191/f3e31a03-7ee5-49c1-ad61-2539d6a789d4)

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/8478191/ee746cc5-d0f9-4f17-b9bd-f90e5f768fd4)

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/8478191/9391fd52-b8b1-45d9-92b9-a370ee2e2b7c)

### NuGet package version

WinUI 3 - Windows App SDK 1.4.1: 1.4.230913002

**Comment by castorix** [comment]
It does not crash on my configuration :

`<WindowsPackageType>None</WindowsPackageType>
<WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>`

Windows 10 22H2
Windows App SDK 1.4.230822000

Test with 1st sample from [MenuFlyout Class](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.menuflyout?view=windows-app-sdk-1.4)

**Comment by JozefGula** [comment]
@castorix can you please try this configuration (?):

```
  <ItemGroup>
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.4.230913002" />
    <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
    <Manifest Include="$(ApplicationManifest)" />
</ItemGroup>
```

**Comment by DarranRowe** [comment]
From what you provided, the stored exception is REGDB_E_CLASSNOTREG. In other words, it is trying to activate an instance but it is unable to because it is unable to find the registration for it.
This is interesting because when building the project, the build process should do everything at once. One suggestion is to clean the solution, make sure that you delete any intermediate files in the solution directory and then build the project again. The reason I mention this is simple. The Windows App SDK is centred around WinRT components. WinRT components are an evolution of COM. This means that these components must be registered in a certain way so the application is able to find them. If you are not using the Windows App SDK self contained, then the application will get all of this information from the Windows App SDK package manifest. If you extract the Windows App Runtime .MSIX file and look at the file named AppxManifest.xml, you can see what is registered.
So, what happens if the application is set to self contained? The WinRT components still need to be registered, so where is this registration information? The build process uses registration free WinRT, which means that it places the class information in the application manifest.
It is very possible that the build process isn't refreshing the application manifest file properly. This could be a problem if you recently moved from an older version of the Windows App SDK to 1.4. Any newer class registrations could be missing. So clear all intermediate files and output, rebuild and then try again. Just in case it needs to be said, treat the output as a unit and don't mix and match files while you are testing.

**Comment by JozefGula** [comment]
Hello @DarranRowe , thank you very much for your response. It seems that this problem is connected to the ".vs" folder. Here are some scenarios: 

1. delete ".vs", "bin", "obj" folders
2. build the project without "WindowsAppSDKSelfContained", Menu MenuFlyout is working properly
3. create "WindowsAppSDKSelfContained" into the project file
4. build the project with "WindowsAppSDKSelfContained", Menu MenuFlyout causes a crash in the application

1. delete ".vs", "bin", "obj" folders
2. build the project with "WindowsAppSDKSelfContained", Menu MenuFlyout is working properly

In the past, I always tried to delete the "bin" and "obj" folders, but I never deleted ".vs" because it is hidden.

It seems that my problem was caused by the fact that I started creating the application without "WindowsAppSDKSelfContained" and when I added it, the ".vs" folder contained something that led to the fact that MenuFlyout caused crashes.

I can definitely confirm that deleting ".vs" and a new build and publishing with "Windows App SDK Self Contained" created an application in which the MenuFlyout also works correctly. Awesome, I am happy.

@DarranRowe thank you very much
